server/channels/api4/preference.go

```diff
@@ -122,7 +123,16 @@ func updatePreferences(c *Context, w http.ResponseWriter, r *http.Request) {
 				return
 			}
 
-			if !c.App.SessionHasPermissionToChannel(c.AppContext, *c.AppContext.Session(), post.ChannelId, model.PermissionReadChannelContent) {
+			channel, ok := channelMap[post.ChannelId]
```

Comment:
Hi @devinbinnie, I think the code does not update the channelMap after fetching the channel, leading to potential repeated fetches for the same channel.